webui/react/src/hooks/useCustomizeColumnsModal.tsx

```diff
@@ -0,0 +1,234 @@
+import { Button, Input, Modal, ModalFuncProps } from 'antd';
+import { ModalFunc } from 'antd/es/modal/confirm';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { FixedSizeList as List } from 'react-window';
+
+import { isEqual } from 'utils/data';
+import { camelCaseToSentence } from 'utils/string';
+
+import css from './useCustomizeColumnsModal.module.scss';
+
+interface ModalState {
+  columns: string[];
+  defaultVisibleColumns: string[];
+  initialVisibleColumns: string[];
+  onSave?: (columns: string[]) => void;
+  visible: boolean;
+}
+
+export interface ShowModalProps {
+  columns: string[];
+  defaultVisibleColumns: string[];
+  initialVisibleColumns: string[];
+  onSave?: (columns: string[]) => void;
+}
+
+interface ModalHooks {
+  showModal: (props: ShowModalProps) => void;
+}
+
+const useCustomizeColumnsModal = (): ModalHooks => {
```

Comment:
**nit:** I know it's a bit wonky but would `useModalCustomizeColumns` be a deal breaker? The idea is to keep all the modal hooks listed near each other in the directory.